sdk/core/Azure.Core/perf/DownloadTest.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Azure.Test.Perf;
+
+namespace Azure.Template.Perf
+{
+    public class DownloadTest : PerfTest<DownloadTestOptions>
+    {
+        // Stream.CopyToAsync default buffer
+        private const int BufferSize = 81920;
+        private readonly HttpPipeline _pipeline;
+        private readonly InProcTestServer _server;
+
+        public DownloadTest(DownloadTestOptions options) : base(options)
+        {
+            var buffer = new byte[BufferSize];
+            _server = InProcTestServer.CreateStaticResponse(options.Size);
+            _pipeline = HttpPipelineBuilder.Build(new DefaultAzureCredentialOptions());
+        }
+
+        public override void Run(CancellationToken cancellationToken)
+        {
+            var message = _pipeline.CreateMessage();
+            message.Request.Uri.Reset(_server.Address);
+
+            _pipeline.Send(message, cancellationToken);
+            message.Response.ContentStream.CopyTo(Stream.Null, BufferSize);
+        }
+
+        public override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            var message = _pipeline.CreateMessage();
```

Comment:
Can `message` be reused across requests?

Reply:
No, it's mutated while request is sent/response received.